app_dart/lib/src/service/scheduler.dart

```diff
@@ -1221,7 +1222,40 @@ $stackTrace
       throw 'No PR found matching this check_run($id, $name)';
     }
 
-    await _runCiTestingStage(pullRequest: pullRequest, checkRunGuard: '$checkRunGuard', logCrumb: logCrumb);
+    try {
+      await _runCiTestingStage(pullRequest: pullRequest, checkRunGuard: '$checkRunGuard', logCrumb: logCrumb);
+    } catch (error, stacktrace) {
+      await githubChecksService.githubChecksUtil.createCheckRun(
```

Comment:
What's the recovery like for this check run? Will it forever remain red? Does it block the PR from being submitted?

Reply:
Before this PR, you get a silent failure and you have to figure it out from scratch. After this PR, you have a bit more information... you could rebase, you could figure out what went wrong, or you could go and find someone for help.

Reply:
Unforseen / unhandled: people clicking rerun on the failed test thinking it would do something. It did, just not what they or I thought. Matan has changed this to a comment.